libbeat/common/match/cmp.go

```diff
@@ -40,13 +40,27 @@ var (
 	patDigits = mustParse(`\d`)
 )
 
+func isRegular(r *syntax.Regexp) bool {
+	const irregular = syntax.FoldCase
+	return (r.Flags & irregular) == 0
+}
+
+func isRegularLiteral(r *syntax.Regexp) bool {
+	return r.Op == syntax.OpLiteral && isRegular(r)
+}
+
+func isSubstringLiteral(r *syntax.Regexp) bool {
```

Comment:
minor: do you need this? just wraps another method. 
if this is for readability i'm ok with it

Reply:
Yeah, it's for readability.